lib/cli.rb

```diff
@@ -0,0 +1,164 @@
+require_relative '../config/environment'
+require 'pry'
+require 'colorize'
+
+
+class CLI
+
+  
+
+  def create_user
+    puts `clear`
+    puts "Please type in your name...then press enter.. I guess..."
+    user_name = STDIN.gets.chomp.strip 
+    $user = User.find_or_create_by(name: user_name)
+  end 
+  
+  def greeting
+    puts `clear`
+    puts "Weather is breezy".colorize(:light_magenta)
+    puts 
+    puts "Weather is cool" 
+    puts 
+    puts "Don't get "+"washed".colorize(:light_magenta).italic+" away by too many choices."
+    puts 
+    puts "~ Select an option below "+"#{$user.name}".colorize(:cyan).italic+" ~"
+    puts 
+  end 
+
+  def cli_program
+    is_running = true 
+    while is_running 
+      puts "1. Search "+ "Weather".colorize(:cyan)
+      puts "2. View Search History"
+      puts "3. Delete History"
+      puts "4. Change a Search Record"
+      puts "5. Exit program"
+
+      choice = STDIN.gets.chomp 
+
+      if choice == "1"
+        puts `clear`
+        Search.all.each do |search|
+          puts "Location: #{search.search_location}"
+        end 
+        puts
+        puts "Please type in a City to see the "+"Weather".colorize(:cyan)+", then press enter" 
+        user_input = STDIN.gets.chomp.strip
+        
+        actual_location = Search.all.find_by(search_location: user_input)
```

Comment:
```suggestion
        actual_location = Search.find_by(search_location: user_input)
```
Unnecessary use of all. Above you were simply iterating over all the instances, whereas here you are doing a find_by, so it already knows you mean to do it on all the instances.